File: javalin/src/main/java/io/javalin/http/ExceptionHandler.java
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */

package io.javalin.http;

import io.javalin.Javalin;
import org.jetbrains.annotations.NotNull;

/**
 * A handler for use with {@link Javalin#exception(Class, ExceptionHandler)}.
 * Is triggered when exceptions are thrown by a {@link Handler}.
 *
 * @see Context
 * @see <a href="https://javalin.io/documentation#exception-mapping">Exception mapping in docs</a>
 */
@FunctionalInterface
public interface ExceptionHandler<T extends Exception> {
    void handle(@NotNull T exception, @NotNull Context ctx);
}

File: javalin/src/main/java/io/javalin/http/ExceptionHandler.kt
/*
 * Javalin - https://javalin.io
 * Copyright 2017 David Åse
 * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
 */
package io.javalin.http

/**
 * A handler for use with [Javalin.exception].
 * Is triggered when exceptions are thrown by a [Handler].
 *
 * @see Context
 *
 * @see [Exception mapping in docs](https://javalin.io/documentation.exception-mapping)
 */
fun interface ExceptionHandler<T : Exception?> {
    fun handle(exception: T, ctx: Context)
}
